### PR TITLE
Added nyaa.si support

### PR DESCRIPTION
**Pull request by datafanatic**
After a couple of days of work, I managed to add nyaa.si support. 

Summary of the changes:
- Nyaa module working;
- Fixed a couple of typos and errors in other modules;
- Updated README.md to reflect the changes;

There are a couple of things missing as of today:
- Searching through categories (mostly implemented but not working 100%);
- Getting the magnet link in the website;
- ~~Colours~~

Other than that, according to the testing I've done, everything works as expected. Talking about tests, perhaps it's time that some are implemented in the project? My module alone added some 200~ LOC, the more modules are added, the harder it will be to test that they all work they way they are supposed to.

### CODE REVIEW

**Comment by kryptxy** [comment]
Thanks! Few changes: 

* Do not hard-code the site URL into main module.  Instead fetch the main site URL from config.ini file, and assign the rest of search parameter to some variable ([this](https://github.com/kryptxy/torrench/blob/master/torrench/modules/thepiratebay.py#L112) might help)

Note: The sites listed in config.ini file are converted to list. This way adding more proxy sites becomes as simple as adding site name in the beginning of list. This conversion takes place in ```Config``` module. This list is then used in the required modules. (See [this](https://github.com/kryptxy/torrench/blob/master/torrench/modules/thepiratebay.py#L29) for refernce).

* Since the module requires config.ini file, Instead of inheriting ```Common``` class, inherit ```Config```. ```Config``` itself inherits ```Common```.

* There is no need for importing beautifulsoup4, requests into this module separately. ```Common``` already provides two handy methods ```http_request()``` and ```http_request_time()```.

* To get colored text, use ```colorify()``` method from ```Common``` module.

**Comment by kryptxy** [comment]
And yes. Tests are much needed. I'll add them.

**Comment by datafanatic** [comment]
> Do not hard-code the site URL into main module. Instead fetch the main site URL from config.ini file, and assign the rest of search parameter to some variable (this might help)

> Since the module requires config.ini file, Instead of inheriting Common class, inherit Config. Config itself inherits Common.

Both easy fixes. I should be able to ship them soon. 

However, regarding `http_requests()` and `http_request_time()` I actually started writing the module by using them, but I was unable to use them for my purposes. When I'm done fixing the aforementioned problems, I'll try using them again. Thanks.

**Comment by datafanatic** [comment]
Okay, I think I've addressed all of your concerns. Please let me know if there's anything else either missing or wrong.

**Comment by kryptxy** [comment]
One thing. Adding more than one proxy to config file won't work. [Here](https://github.com/kryptxy/torrench/pull/21/commits/b82cc6e442cd7410a42dbd05954c04dbe1323894#diff-7622671d6091f0bf1c51f3d80ad5d266R35) we see that only the first site is tested. In future if this site goes down, the rest of the proxies (when added) would not be considered. 

See how [skytorrents](https://github.com/kryptxy/torrench/blob/master/torrench/modules/skytorrents.py#L52) resolve this. This eliminates the restriction of using only the first site present in the list. 

It's better to implement this now, as adding proxies in future would be simple.

**Comment by datafanatic** [comment]
Done. Please review the changes.

**Comment by kryptxy** [comment]
I'll check it by today

**Comment by kryptxy** [comment]
@datafanatic I sent a few modifications (twitter DM). Thoughts?

File: torrench/modules/nyaa.py [hunk 1]
```diff
@@ -184,7 +154,7 @@ def fetch_results(self):
             self.logger.debug("No results were found for `%s`.", self.title)
             return -1
         self.logger.debug("Results fetched. Showing table.")
-        self.mapper.insert(self.index+1, (name, urls))
+        self.mapper.insert(self.index+1, (name, urls, magnets))
         return list(zip(name, ["--"+str(idx)+"--" for idx in range(self.index)], sizes, seeds, leeches))
```
Comment (kryptxy) [inline torrench/modules/nyaa.py]: Here, set ```range(self.index)]``` to ```range(1, self.index+1)]```

File: torrench/modules/nyaa.py [hunk 2]
```diff
@@ -193,14 +163,30 @@ def select_torrent(self):
...
+                prompt = int(input("(0 to exit)\nIndex > "))
+                self.logger.debug("Selected index {idx}".format(idx=prompt))
                 if prompt == 0:
                     print("Bye!")
                     break
                 else:
-                    selected_index, download_url = self.mapper[0][0][prompt], self.mapper[0][1][prompt]
-                    print("Downloading: \n" + selected_index)
-                    self.get_torrent(download_url, selected_index)
+                    selected_index, download_url, magnet_url = self.mapper[0][0][prompt], self.mapper[0][1][prompt], self.mapper[0][2][prompt]
```
Comment (kryptxy) [inline torrench/modules/nyaa.py]: And here: set ```[prompt]``` to ```[prompt-1]```

**Comment by kryptxy** [comment]
PR Merged!

**Pull request merged**